### PR TITLE
Image promotion for kube-state-metrics v2.8.2

### PR DESCRIPTION
**Pull request by mrueg**
@rexagod @dgrisonnet @fpetkovski

### CODE REVIEW

**Comment by BenTheElder** [comment]
Hey, you'll have to move this to registry.k8s.io/, sorry we're working on https://github.com/kubernetes/enhancements/issues/3720 ref https://github.com/kubernetes/k8s.io/pull/4934 https://github.com/kubernetes/test-infra/pull/29069 

[forcing bot to process this, will reopen immediately]
/close

**Comment by mrueg** [comment]
@BenTheElder since we would like to release tomorrow and the k8s.gcr.io freeze date is in 2 weeks, can we still process this one?

**Comment by BenTheElder** [comment]
> @BenTheElder since we would like to release tomorrow and the k8s.gcr.io freeze date is in 2 weeks, can we still process this one?

#4934 just moved the configurations from k8s.gcr.io/ to registry.k8s.io/

If you update this PR to move identically you're good to go. But no files will be under k8s.gcr.io/ anymore

That's just a rename to clarify things as part of the freeze, it's not strictly coupled to which registries the images wind up in.

However: Please DO NOT tell people to pull your new image from k8s.gcr.io even if it winds up available there. k8s.gcr.io is deprecated and https://registry.k8s.io is the way forward for the project. We have a tracking issue to ensure all subproject references get updated independent of the freeze on GCR https://github.com/kubernetes/k8s.io/issues/4738

File: k8s.gcr.io/images/k8s-staging-kube-state-metrics/images.yaml
Comment (BenTheElder) [inline k8s.gcr.io/images/k8s-staging-kube-state-metrics/images.yaml:1]: more specifically: rebase or recreate this diff against `registry.k8s.io/images/k8s-staging-kube-state-metrics/images.yaml` and the bot will lift the `do-not-merged/blocked-paths` label.

the same tooling will copy the images to production, and they'll be at pullable at `registry.k8s.io/$image:$tag` (as they also would have been before moving the source directory here)

**Pull request closed without merging**